src/components/pages/Volunteers/index.tsx

```diff
@@ -79,20 +107,69 @@ const UnconnectedVolunteers: React.FC<PropsFromRedux> = ({
   };
 
   useEffect(() => {
-    if (!hasFetchedVolunteers) {
-      loadVolunteers();
+    if (
+      !hasFetchedVolunteers &&
+      volunteers.all_volunteers.length === 0 &&
+      org
+    ) {
+      loadVolunteers(token, org.id);
       setHasFetchedVolunteers(true);
     }
-
-    const results = volunteers.filter((volunteer) =>
+    const results = volunteers.all_volunteers.filter((volunteer) =>
       volunteer.name.toLowerCase().includes(searchQuery.toLowerCase()),
     );
     setFilteredVolunteers(results);
-  }, [hasFetchedVolunteers, loadVolunteers, volunteers, searchQuery]);
+  }, [
+    hasFetchedVolunteers,
+    loadVolunteers,
+    volunteers.all_volunteers,
+    searchQuery,
+  ]);
+
+  if (!user.authInfo.isLoggedIn) {
+    loading();
+    return <Redirect to="/login" />;
+  }
+
+  if (volunteers.error.message === 'Expired Token') {
```

Comment:
qq - I know we're using this condition in a lot of places. Would it make sense to create an enum with the error messages 

OR just create a util function isExpired(volunteers.error) and just reuse it